gap/attr.gi

```diff
@@ -369,6 +369,181 @@ function(D)
 end
 );
 
+BindGlobal("DIGRAPHS_ChromaticNumberZykov",
+function(D)
+  local nr, ZykovReduce, chrom;
+  nr := DigraphNrVertices(D);
+  # Recursive function call
+  ZykovReduce := function(D)
+    local nr, D_contract, adjacent, vertices, v, x, y, x_i, y_i, found, deg;
+    nr := DigraphNrVertices(D);
+    # Update upper bound if possible.
+    chrom := Minimum(nr, chrom);
+    # Leaf nodes are either complete graphs or q-cliques. The chromatic number
+    # is then the smallest q-clique found.
+    if not IsCompleteDigraph(D) and IsEmpty(CliquesFinder(D, fail, [], 1, [],
+                                                          [], false, chrom,
+                                                          true)) then
+      # Get adjacency function
+      adjacent := DigraphAdjacencyFunction(D);
+      # Sort vertices by degree, so that higher degree vertices are picked first
+      vertices := [1 .. nr];
+      deg   := ShallowCopy(OutDegrees(D));
+      SortParallel(deg, vertices, {x, y} -> x > y);
+      # Choose two non-adjacent vertices x, y
+      # This is just done by ascending ordering.
+      found := false;
+      for x_i in [1 .. nr] do
+        x := vertices[x_i];
+        for y_i in [x_i + 1 .. nr] do
+          y := vertices[y_i];
+          if not adjacent(x, y) then
+            found := true;
+            break;
+          fi;
+        od;
+        if found then
+          break;
+        fi;
+      od;
+      Assert(1, x <> y, "x and y must be different");
+      Assert(1, found, "No adjacent vertices");
+      # Colour the vertex contraction.
+      # A contraction of a graph effectively merges two non adjacent vertices
+      # into a single new vertex with the edges merged.
+      # We merge y into x, keeping x.
+      D_contract := DigraphMutableCopy(D);
```

Comment:
Looking through the manual I can't seem to find any reference to `DigraphContractEdge`, but I believe it would be possible to replace the step with `DigraphQuotient` by taking the partition the two adjacent vertices in one part, and then the remaining vertices are singleton parts. I need to look into the implementation of `DigraphQuotient`, as it seems more general that what is needed here and so could be slower.

Reply:
I see now there is an open pull request for `DigraphContractEdge`, so I'll use that once the required changes are made.